src/components/slider/slider.ts

```diff
@@ -129,13 +132,18 @@ export class MdSlider implements AfterContentInit {
     this.isDragging = false;
     this._renderer.addFocus();
     this.updateValueFromPosition(event.clientX);
+
+    // Once the click is over the thumb has to snap to its new physical location.
+    this.updatePercentFromValue();
+    this._renderer.updateThumbAndFillPosition(this._percent, this._sliderDimensions.width);
```

Comment:
Nit: Since it seems like you'll want to call these two together when you want to snap (here and in line 169), you might consider breaking it out into a function like `snaptoValue()` or something.